src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -719,5 +732,20 @@ internal static unsafe partial class Utf8Utility
             scalarCountAdjustment = tempScalarCountAdjustment;
             return pInputBuffer;
         }
+
+        // Passing the values of initialMask and mostSignificantBitMask is slightly more efficient than using static readonly fields as of right now.
+        // The expected values are:
+        // initialMask = 0x1001 = 0001 0000 0000 0001 (2 bytes)
+        // mostSignficantBitMask = 0x80 = 1000 0000 (1 byte)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static ulong GetNonAsciiBytes(Vector128<byte> value, Vector128<byte> initialMask, Vector128<byte> mostSignificantBitMask)
+        {
+            Debug.Assert(AdvSimd.Arm64.IsSupported);
+
+            Vector128<byte> mostSignificantBitIsSet = AdvSimd.CompareEqual(AdvSimd.And(value, mostSignificantBitMask), mostSignificantBitMask);
+            Vector128<byte> extractedBits = AdvSimd.And(mostSignificantBitIsSet, initialMask);
```

Comment:
This will also have to be updated to either return the precise Sse2 mask, or you could move `BitOperations.TrailingZeroCount` here and simply return the index of the first non-ascii element.

Reply:
I'll try to do the second option, since there's only one place where this method is called.

Reply:
Keep in mind you'd need to extract the equivalent logic for Sse2, so that the function call in line 168 is removed.

Reply:
Yes nevermind, I used the AddPairwise version of the method. Thanks.